### ISSUE TITLE
fix grouped nanmean with bool

### ISSUE THREAD

**Issue by dcherian** [numbagg/numbagg#131]
```python
numbagg.grouped.group_nanmean(np.array([True, True, False]), [0, 0, 0], axis=-1)
# array([0], dtype=int32)
np.nanmean(np.array([True, True, False]))
# 0.6666666666666666
```

Should I just promote these to float instead?

**Comment by dcherian** [comment]
Another edge case

```
numbagg.grouped.group_nansum(np.array([True, True, False]), [0, 0, 0], axis=-1).dtype
# int32
np.nansum(np.array([True, True, False]), axis=-1).dtype
# int64
```

**Comment by max-sixty** [comment]
This is somewhat downstream of #121. There's a bullet tucked away there "Always keep types — current state (except for bools...)"

The current state is that we maintain types, apart from a quirk where we convert bools to int32.

I think the closest thing would be for flox to convert into whatever type it wants before submitting, and then isn't dependent on any handling in numbagg. Then later we can optimize by solving #121. WDYT? Or is this more complicated than converting bools to floats and flox needs to then handle lots of edge cases?

**Comment by dcherian** [comment]
Yeah, I'll just promote for `nanmean` now. I think that's easiest.
